COLLADAMaya/src/COLLADAMayaDocumentExporter.cpp

```diff
@@ -335,6 +335,37 @@ namespace COLLADAMaya
             COLLADASW::URI sourceFileUri ( COLLADASW::URI::nativePathToUri ( currentScene ) );
             if ( sourceFileUri.getScheme ().empty () )
                 sourceFileUri.setScheme ( COLLADASW::URI::SCHEME_FILE );
+			if (sourceFileUri.scheme() == COLLADASW::URI::SCHEME_FILE && ExportOptions::sourceDataFormat() != ExportOptions::Absolute)
+			{
+				COLLADABU::URI relativeTo;
+				switch (ExportOptions::sourceDataFormat())
+				{
+					case ExportOptions::RelativeToExportedDAE:
+					{
+						relativeTo = COLLADABU::URI::nativePathToUri(getFilename());
+						break;
+					}
+					case ExportOptions::RelativeToMayaProjectWorkspace:
+					{
+						MString workspace;
+						if (MGlobal::executeCommand(MString("workspace -q -rd;"), workspace))
```

Comment:
I see. Well the `xml:base` attribute can be applied to an <element>, but that would not be valid in COLLADA 1.4 or 1.5, since it's not declared in the schema for the `<source_data>` element. That would be a good revision to the schemata.

I think best solution under the circumstances is that you can add `source_data_base_uri=file:///path/to/asset/source` as a pseudo-attribute in the `<comments>` elements. Use the value if present, otherwise relative url must follow the RFC rules.